weave/trace_server/trace_server_interface.py

```diff
@@ -318,6 +319,75 @@ class FileContentReadRes(BaseModel):
     content: bytes
 
 
+class FeedbackPayloadReactionReq(BaseModel):
+    emoji: str
+
+
+class FeedbackPayloadNoteReq(BaseModel):
+    note: str = Field(min_length=1, max_length=1024)
+
+
+class FeedbackCreateReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    weave_ref: str = Field(examples=["weave:///entity/project/object/name:digest"])
+    creator: typing.Optional[str] = Field(default=None, examples=["Jane Smith"])
+    feedback_type: str = Field(examples=["custom"])
+    payload: typing.Dict[str, typing.Any] = Field(
+        examples=[
+            {
+                "key": "value",
+            }
+        ]
+    )
+
+
+class FeedbackCreateReqForInsert(FeedbackCreateReq):
+    wb_user_id: str
+
+
+# The response provides the additional fields needed to convert a request
+# into a complete Feedback.
+class FeedbackCreateRes(BaseModel):
+    id: str
+    created_at: datetime.datetime
+    wb_user_id: str
+    payload: typing.Dict[str, typing.Any]  # If not empty, replace payload
+
+
+class Feedback(FeedbackCreateReqForInsert):
+    id: str
+    created_at: datetime.datetime
+
+
+class FeedbackQueryReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    fields: typing.Optional[list[str]] = Field(
+        default=None, examples=[["id", "feedback_type", "payload.note", "count(*)"]]
+    )
+    query: typing.Optional[Query] = None
+    # TODO: I think I would prefer to call this order_by to match SQL, but this is what calls API uses
+    # TODO: Might be nice to have shortcut for single field and implied ASC direction
+    # TODO: I think _SortBy shouldn't have leading underscore
+    sort_by: typing.Optional[typing.List[_SortBy]] = None
+    limit: typing.Optional[int] = Field(default=None, examples=[10])
+    offset: typing.Optional[int] = Field(default=None, examples=[0])
+
+
+class FeedbackQueryRes(BaseModel):
+    # Note: this is not a list of Feedback because user can request any fields.
```

Comment:
interesting that we support cases where the fields are not present. alternative idea for the "fields" in the request to only control payload fields, which would result in a list of value feedbacks.

Reply:
In the case of feedback there are probably not a lot of cases (although supporting COUNT(*) is already useful) where you wouldn't want the payload, but if think about API patterns for more complicated objects, like calls, I think there are more cases where you want the ability to only pull down the fields you need.

Reply:
Definitely agree that it is useful to filter fields. However, in practice, there are a bunch of id/date/name like fields that are low cost. Then there are big ones (inputs, output, attributes, summary). Similarly here you have a few simple fields, then a possibly huge field (payload). My mental model for calls has been to always return the "simple fields". Then essentially have "partials" for the others. In such cases, the typing would be consistent because the big fields are still dicts, just empty.

I don't think this is a blocking comment btw, but maybe something to consider (both ways) as we start to harden the API